OrbitGl/CaptureSerializer.h

```diff
@@ -14,31 +15,84 @@
 
 #include "CaptureData.h"
 #include "OrbitBase/Result.h"
+#include "TimeGraph.h"
 #include "capture_data.pb.h"
 
 class CaptureSerializer {
  public:
-  void Save(std::ostream& stream, const CaptureData& capture_data);
-  ErrorMessageOr<void> Save(const std::string& filename, const CaptureData& capture_data);
+  template <class TimersIterator>
+  static ErrorMessageOr<void> Save(
+      const std::string& filename, const CaptureData& capture_data,
+      const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
+      TimersIterator timers_iterator_begin, TimersIterator timers_iterator_end);
 
   ErrorMessageOr<void> Load(std::istream& stream);
   ErrorMessageOr<void> Load(const std::string& filename);
 
-  class TimeGraph* time_graph_;
+  TimeGraph* time_graph_;
 
   static bool ReadMessage(google::protobuf::Message* message,
                           google::protobuf::io::CodedInputStream* input);
   static void WriteMessage(const google::protobuf::Message* message,
                            google::protobuf::io::CodedOutputStream* output);
 
  private:
-  orbit_client_protos::CaptureInfo GenerateCaptureInfo(const CaptureData& capture_data);
+  template <class TimersIterator>
+  static void Save(std::ostream& stream, const CaptureData& capture_data,
+                   const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
+                   TimersIterator timers_iterator_begin, TimersIterator timers_iterator_end);
+
+  static orbit_client_protos::CaptureInfo GenerateCaptureInfo(
+      const CaptureData& capture_data,
+      const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map);
+
   [[nodiscard]] CaptureData GenerateCaptureData(
       const orbit_client_protos::CaptureInfo& capture_info);
 
-  orbit_client_protos::CaptureHeader header;
+  orbit_client_protos::CaptureHeader header_;
 
-  const std::string kRequiredCaptureVersion = "1.52";
+  static inline const std::string kRequiredCaptureVersion = "1.52";
 };
 
+template <class TimersIterator>
+void CaptureSerializer::Save(std::ostream& stream, const CaptureData& capture_data,
+                             const absl::flat_hash_map<uint64_t, std::string>& key_to_string_map,
+                             TimersIterator begin, TimersIterator end) {
```

Comment:
Maybe rename this to _timer_info_begin_ and _timer_info_end_ or _timers_begin_ or sth like that for more clarity when reading the code below

Reply:
Or _timers_iterator_begin_ and  _timers_iterator_end_ like in the method below

Reply:
Thanks. I made naming consistent.

Reply:
Thanks!